src/main/java/lotto/domain/BonusNumber.java

```diff
@@ -1,34 +1,30 @@
 package lotto.domain;
 
 import lotto.domain.lotto.LottoNumber;
-import lotto.exception.InValidBonusNumberException;
 
 import java.util.List;
 
 public class BonusNumber {
     private final LottoNumber bonusNumber;
 
-    public BonusNumber(LottoNumber bonusNumber) {
+    private BonusNumber(LottoNumber bonusNumber) {
         this.bonusNumber = bonusNumber;
     }
 
-    private BonusNumber(LottoNumber bonusNumber, WinningNumbers winningNumbers) {
-        validateBonusNumberWithOutWinningNumbers(bonusNumber, winningNumbers);
-        this.bonusNumber = bonusNumber;
+    public static BonusNumber valueOf(LottoNumber bonusNumber) {
+        return new BonusNumber(bonusNumber);
     }
 
-    private void validateBonusNumberWithOutWinningNumbers(LottoNumber bonusNumber, WinningNumbers winningNumbers) {
-        if (winningNumbers.contains(bonusNumber)) {
-            throw new InValidBonusNumberException(bonusNumber.lottoNumber());
-        }
+    public boolean isMatched(List<LottoNumber> lottoNumbers) {
+        return lottoNumbers.stream()
+                .anyMatch(bonusNumber::equals);
```

Comment:
Nit: `contains()`로 처리해볼 수도 있겠네요.
```suggestion
        return lottoNumbers.contains(bonusNumber);
```